src/ServicePulse.Host/vue/src/components/failedmessages/DeletedMessageGroups.vue

```diff
@@ -144,38 +162,37 @@ function createWorkflowState(optionalStatus, optionalTotal, optionalFailed) {
 }
 
 //Restore operation
-function showRestoreGroupDialog(group) {
+function showRestoreGroupDialog(group: ExtendedFailureGroupView) {
   groupRestoreSuccessful.value = null;
   selectedGroup.value = group;
   showRestoreGroupModal.value = true;
 }
 
 async function restoreGroup() {
-  // We're starting a restore, poll more frequently
-  changeRefreshInterval(1000);
-
-  selectedGroup.value = archiveGroups.value.find((group) => group.id === selectedGroup.value.id);
-
-  undismissedRestoreGroups.value.push(selectedGroup.value);
-
   const group = selectedGroup.value;
-  group.workflow_state = { status: "restorestarted", message: "Restore request initiated..." };
-  group.operation_start_time = new Date().toUTCString();
-
-  const result = await useRestoreGroup(group.id);
-  if (result.message === "success") {
-    groupRestoreSuccessful.value = true;
-    useShowToast("info", "Info", "Group restore started...");
-  } else {
-    groupRestoreSuccessful.value = false;
-    useShowToast("error", "Error", "Failed to restore the group:" + result.message);
+  if (group) {
+    // We're starting a restore, poll more frequently
+    changeRefreshInterval(1000);
+    undismissedRestoreGroups.value.push(group);
+
+    group.workflow_state = { status: "restorestarted", message: "Restore request initiated..." };
+    group.operation_start_time = new Date().toUTCString();
+
+    const result = await useRestoreGroup(group.id);
+    if (result.message === "success") {
```

Comment:
This looks weird

Reply:
yep, pretty terrible:
```js
async function evaluateResponse(response: Response) {
  return {
    message: response.ok ? "success" : `error: ${response.statusText}`,
  };
}```

I'll see how painful it is to refactor all the usages of this